python_modules/dagster/dagster/_core/execution/submit_asset_runs.py

```diff
@@ -0,0 +1,304 @@
+
+
+def submit_asset_run(
+    run_id: Optional[str],
+    run_request: RunRequest,
+    run_request_index: int,
+    instance: DagsterInstance,
+    workspace_process_context: IWorkspaceProcessContext,
+    asset_graph: ExternalAssetGraph,
+    run_request_execution_data_cache: Dict[int, RunRequestExecutionData],
+    debug_crash_flags: SingleInstigatorDebugCrashFlags,
+    logger: logging.Logger,
+) -> DagsterRun:
+    """Submits a run for a run request that targets an asset selection. If the run already exists,
+    submits the existing run. If the run does not exist, creates a new run and submits it, ensuring
+    that the created run targets the given asset selection.
+    """
+    check.invariant(not run_request.run_config, "Asset run requests have no custom run config")
+    asset_keys = check.not_none(run_request.asset_selection)
+
+    check.invariant(len(asset_keys) > 0)
+
+    # check if the run already exists
+    existing_run = instance.get_run_by_id(run_id) if run_id else None
+    if existing_run:
+        if existing_run.status != DagsterRunStatus.NOT_STARTED:
+            logger.warn(
+                f"Run {run_id} already submitted on a previously interrupted tick, skipping"
+            )
+
+            check_for_debug_crash(debug_crash_flags, "RUN_SUBMITTED")
+            check_for_debug_crash(debug_crash_flags, f"RUN_SUBMITTED_{run_request_index}")
+
+            return existing_run
+        else:
+            logger.warn(
+                f"Run {run_id} already created on a previously interrupted tick, submitting"
+            )
+            run_to_submit = existing_run
+    else:
+        run_to_submit = _create_asset_run(
+            run_id,
+            run_request,
+            run_request_index,
+            instance,
+            run_request_execution_data_cache,
+            asset_graph,
+            workspace_process_context,
+            debug_crash_flags,
+            logger,
+        )
+
+    instance.submit_run(run_to_submit.run_id, workspace_process_context.create_request_context())
+
+    check_for_debug_crash(debug_crash_flags, "RUN_SUBMITTED")
+    check_for_debug_crash(debug_crash_flags, f"RUN_SUBMITTED_{run_request_index}")
+
+    asset_key_str = ", ".join([asset_key.to_user_string() for asset_key in asset_keys])
+
+    logger.info(
+        f"Submitted run {run_to_submit.run_id} for assets {asset_key_str} with tags"
+        f" {run_request.tags}"
+    )
+
+    return run_to_submit
+
+
+def submit_asset_runs_in_chunks(
+    run_requests: Sequence[RunRequest],
+    reserved_run_ids: Optional[Sequence[str]],
+    chunk_size: int,
+    instance: DagsterInstance,
+    workspace_process_context: IWorkspaceProcessContext,
+    asset_graph: ExternalAssetGraph,
+    debug_crash_flags: SingleInstigatorDebugCrashFlags,
+    logger: logging.Logger,
+) -> Iterator[Optional[Sequence[Tuple[RunRequest, DagsterRun]]]]:
+    """Submits runs for a sequence of run requests that target asset selections in chunks. Yields
```

Comment:
the AMP callsite isn't getting much benefit from the chunk part of this method right? chunk_size being 1 and all? An alternative would be to have it call some other public function that AMP calls and leave the chunking callsite to just be used by backfills (in case we add more chunk-specific stuff in the future, say logging, that might not apply to AMP)